### DISCUSSION TITLE
SRT Encryption - Pass Phrase

### DISCUSSION THREAD

**Discussion by bungleweed** [OvenMediaLabs/OvenMediaEngine#1200] [Ideas] [answered]
Hi,

When using SRT to provide stream source to OME is it possible to configure a passphrase for the encryption in the config? I want to send video using SRT into OME and ensure it is encrypted in transit. I'm not sure if it is encrypted is a passphrase is not able to be specified in the Server.xml when using SRT as the Live Source protocol. I am able to send a stream to OME using SRT fine but I'm not sure its encrypted without a passphras/password being specified. Is this supported in the config as it is not documented from what I can see. My encoder support a SRT password option (it's a Uray encoder and the option to specity the SRT encryption password is there). I need to know what to put in the Server.xml config to match it up.

**Comment by bungleweed** [comment]
Anyone help with this at all?

**Comment by getroot** [comment]
Currently it is not possible to set passphrase for SRT in Server.xml. This may be supported in the future. To make this possible now, you have to write it yourself in the source code.

[a comment by an automated account is not shown]

**Comment by dimiden** [comment, marked as answer]
@bungleweed 
I just added a feature to enable setting a passphrase (https://github.com/AirenSoft/OvenMediaEngine/commit/09ce8b313ce9633fd176bb3e6930ad253908d417).
To use this feature, please configure it as follows inside Server/Bind/Providers/SRT:

```
<SRT>
	<Options>
		<Option>
			<Key>SRTO_PBKEYLEN</Key>
			<Value>16</Value>
		</Option>
		<Option>
			<Key>SRTO_PASSPHRASE</Key>
			<Value>your_pass_phrase</Value>
		</Option>
	</Options>
</SRT>
```

Then, you can publish your stream by using a SRT URL like this: ```srt://<YOUR_OME_IP>:9999?streamid=srt%3A%2F%2F<YOUR_OME_IP>%3A9999%2Fapp%2Fstream&passphrase=your_pass_phrase&pbkeylen=16```

**Reply by bungleweed** [reply]
This is awesome. Thanks so much for sorting!

**Comment by bchah** [comment]
@dimiden Thank you for this addition! It is a small but powerful feature that many users will be thrilled to hear about. I can also confirm it is working perfectly 👍🏻 Excellent work!

**Comment by ericsouza** [comment]
hey @dimiden 👋🏻

hope this message finds you well

Do you know if OvenMediaEngine supports rotating the SRT passphrase without interrupting active live streams? I’m thinking of a scenario where the passphrase might be compromised. Maybe something like having two passphrases and handle like this:

1. Keep the current passphrase for existing connections
2. Use a new passphrase for any new connections

Once no active streams are using the old one, we’d rotate and making the new passphrase the current, and generate another for future connections.
